cairo-contracts/packages/apps/src/transfer/components/transfer.cairo

```diff
@@ -277,6 +277,29 @@ pub mod TokenTransferComponent {
         }
     }
 
+    // -----------------------------------------------------------
+    // ICreateToken
+    // -----------------------------------------------------------
+
+    #[embeddable_as(CreateIbcToken)]
+    impl CreateTokenImpl<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>,
+    > of ICreateIbcToken<ComponentState<TContractState>> {
+        fn create_ibc_token(
+            ref self: ComponentState<TContractState>,
+            chan_id_on_b: ChannelId,
```

Comment:
We should also check that the given channel ID is valid, and there is already a corresponding channel that is opened.